preprocessing/cameras/create_video_segments_with_audio_per_participant.py

```diff
@@ -0,0 +1,306 @@
+            the few reference values manually extracted for conflab and hardcoded in constants.py.
+            This may be an issue with ffmpeg/ffprobe, and this problem has been observed in a subset
+            of PCs, but not in all of them. Try fixing your system or try another PC. We have no
+            clue what is causing this issue!!
+            Extracted: {timecodes_of_4G_raw_videos_for_cam_index[camera_index][video_index]} vs Reference: {reference_timecode}
+            """
+
+    # We create the folder for the audio segments per participant
+    audio_segments_per_participant_folder = (
+        AUDIO_SEGMENTS_PER_PARTICIPANT_FOLDER_FOR_ALL_CAMS_IN_LOCAL
+        / f"cam{camera_index}"
+    )
+    audio_segments_per_participant_folder.mkdir(parents=True, exist_ok=True)
+
+    video_with_audio_or_only_audio_subfolder = (
+        "video_with_audio"
+        if not PRODUCE_VIDEO_PER_PARTICIPANT_WITH_BLACK_VIDEO
+        else "only_audio"
+    )
+    video_segments_with_participant_audio_for_camera_folder_path = (
+        VIDEO_SEGMENTS_FOLDER_IN_LOCAL
+        / video_with_audio_or_only_audio_subfolder
+        / f"cam{camera_index}"
+    )
+    video_segments_with_participant_audio_for_camera_folder_path.mkdir(
+        parents=True, exist_ok=True
+    )
+
+    # Now we traverse raw videos of 4G, associated to the given camera_index
+    for video_index, raw_video_file_basename in tqdm(
+        camera_raw_to_segment[f"cam{camera_index}"].items(),
+        desc=f"-Raw videos on cam{camera_index}",
+        leave=False,
+        disable=not DISPLAY_PROGRESS_BARS,
+    ):
+        raw_video_basename = camera_raw_to_segment[f"cam{camera_index}"][video_index]
+        # Now we traverse over the video segments, i.e., the videos clipped in 2 minutes
+        # segments, extracted from each raw video segment of 4G
+        for segment_index in tqdm(
+            range(1, 10),
+            desc=f"--Segments on {raw_video_basename}",
+            leave=False,
+            disable=not DISPLAY_PROGRESS_BARS,
+        ):
+            video_segment_file_basename = (
+                f"vid{video_index}-seg{segment_index}-scaled-denoised.mp4"
+            )
+            # We get the video path. First from the local storage, which is faster, and if
+            # it does not exists (for example, it hasn't been copied) we get it from staff
+            # bulk storage
+            video_segment_path = (
+                VIDEO_SEGMENTS_FOLDER_IN_LOCAL
+                / f"cam{camera_index}"
+                / video_segment_file_basename
+            )
+            if not video_segment_path.exists():
+                video_segment_path = (
+                    VIDEO_SEGMENTS_FOLDER_IN_STAFF_BULK
+                    / f"cam{camera_index}"
+                    / video_segment_file_basename
+                )
+            #
+
+            # The start of the segment is the the start of the video plus the delta for the segment
```

Comment:
Outdated comment

Reply:
Fixed in 124b5be58b35d448a82a245bb968c691206c83f8